bitmovin-analytics-datasource/src/types/queryFilter.ts

```diff
@@ -0,0 +1,37 @@
+import { SelectableValue } from '@grafana/data';
+import { QueryAdAttribute } from './queryAdAttributes';
+import { QueryAttribute } from './queryAttributes';
+
+export enum QUERY_FILTER_OPERATORS {
+  GT = 'GT',
+  GTE = 'GTE',
+  LT = 'LT',
+  LTE = 'LTE',
+  EQ = 'EQ',
+  NE = 'NE',
+  CONTAINS = 'CONTAINS',
+  NOTCONTAINS = 'NOTCONTAINS',
+  IN = 'IN',
+}
+
+export type QueryFilterOperator = keyof typeof QUERY_FILTER_OPERATORS;
+
+export const SELECTABLE_QUERY_FILTER_OPERATORS: Array<SelectableValue<QueryFilterOperator>> = [
+  { value: QUERY_FILTER_OPERATORS.GT, label: QUERY_FILTER_OPERATORS.GT },
+  { value: QUERY_FILTER_OPERATORS.GTE, label: QUERY_FILTER_OPERATORS.GTE },
+  { value: QUERY_FILTER_OPERATORS.LT, label: QUERY_FILTER_OPERATORS.LT },
+  { value: QUERY_FILTER_OPERATORS.LTE, label: QUERY_FILTER_OPERATORS.LTE },
+  { value: QUERY_FILTER_OPERATORS.EQ, label: QUERY_FILTER_OPERATORS.EQ },
+  { value: QUERY_FILTER_OPERATORS.NE, label: QUERY_FILTER_OPERATORS.NE },
+  { value: QUERY_FILTER_OPERATORS.CONTAINS, label: QUERY_FILTER_OPERATORS.CONTAINS },
+  { value: QUERY_FILTER_OPERATORS.NOTCONTAINS, label: QUERY_FILTER_OPERATORS.NOTCONTAINS },
+  { value: QUERY_FILTER_OPERATORS.IN, label: QUERY_FILTER_OPERATORS.IN },
+];
+
+export type QueryFilter = {
+  name: QueryAdAttribute | QueryAttribute;
+  operator: QueryFilterOperator;
+  value: QueryFilterValue;
+};
```

Comment:
According to our api there is another property the filter can have. It's the `field` property where a reference to different data column can be set
https://github.com/bitmovin-engineering/analytics-resource-service/blob/b5aae2eb4d09275f75e80e8ace301979a711c49c/src/main/java/com/bitmovin/api/analytics/rest/dtos/request/QueryFilter.kt#L14-L15

Though i don't know if it is necessary to include this here.

Reply:
We didn't include it in the old plugin, so for the first updated version I won't include it. But I will create a ticket to investigate if we should put it in. 👍